.github/workflows/spectral-lint.yml

```diff
@@ -0,0 +1,27 @@
+name: Spectral Lint Commited OpenAPI Spec
+
+# Trigger the workflow on pull requests and pushes to the main branch
+on:
+  pull_request:
+    paths: 
+      - 'tools/spectral/**'
+      - 'openapi/**.yaml'
+  push:
+    branches:
+      - main
+
+jobs:
+  spectral-lint:
+    runs-on: ubuntu-latest
+    
+    steps:
+      # Checkout the code
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
```

Comment:
Can we just use https://github.com/stoplightio/spectral-action?

Reply:
updated